README.md

```diff
@@ -5,5 +5,121 @@
 <picture>
   <source media="(prefers-color-scheme: dark)" srcset="https://github.com/Hyp-ed/hyped-2024/assets/43144010/12892983-b036-4ec3-b624-1c997f85bf94">
   <source media="(prefers-color-scheme: light)" srcset="https://github.com/Hyp-ed/hyped-2024/assets/43144010/54f3db17-be2b-4473-a963-b7d7d8c24a9a">
-  <img alt="The HYPED logo." style="margin:100px" src="[https://user-images.githubusercontent.com/25423296/163456779-a8556205-d0a5-45e2-ac17-42d089e3c3f8.png](https://github.com/Hyp-ed/hyped-2024/assets/43144010/54f3db17-be2b-4473-a963-b7d7d8c24a9a)">
+  <img alt="The HYPED logo." src="[https://user-images.githubusercontent.com/25423296/163456779-a8556205-d0a5-45e2-ac17-42d089e3c3f8.png](https://github.com/Hyp-ed/hyped-2024/assets/43144010/54f3db17-be2b-4473-a963-b7d7d8c24a9a)">
 </picture>
+
+&nbsp;
+
+![Build Shield](https://github.com/Hyp-ed/hyped-2025/actions/workflows/ci.yml/badge.svg) ![TODO Shield](https://img.shields.io/github/search/hyp-ed/hyped-2025/TODOLater?color=red&label=TODO%20counter)
+
+## Software Architecture
+
+This repository contains the software for the HYPED 2025 pod. The software is divided into two main components:
+
+### 1. Pod-side Code
+
+The pod-side code is responsible for controlling the pod's systems, including the motor & levitation controllers, sensors, localisation system, and communication with the base station. The pod-side code is written in Rust using [Embassy](https://embassy.dev/) and runs on the pod's STM32 microcontrollers.
+
+We use the following microcontrollers on our pod:
+
+- [STM32L476RG](https://www.st.com/en/microcontrollers-microprocessors/stm32l476rg.html)
+- [STM32F767ZI](https://www.st.com/en/microcontrollers-microprocessors/stm32f767zi.html)
+- [STM32H743](https://www.st.com/en/microcontrollers-microprocessors/stm32h743zi.html)
+- [STM23L432](https://www.st.com/en/microcontrollers-microprocessors/stm32l432kc.html)
+
+| Sub-system         | Microcontroller | Number |
+| ------------------ | --------------- | ------ |
+| Localisation       | STM32H743       | 1      |
+| Telemetry          | STM32H743       | 1      |
+| Levitation control | STM32F767       | 8      |
```

Comment:
Won't this be 6? or are you controlling opposing horizontal lev units independently?

Reply:
honestly not sure, will leave it TBD for now